### PR TITLE
Fix prefetching messages in background marking them as read

### LINKED ISSUE #822: Prefetching messages in background marks them as read
As noticed by @DeepDiver1975, messages are marked as read once they are fetched in background

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@jancborchardt as discussed

**Comment by jancborchardt** [comment]
@Xenopathic @ChristophWurst @Gomez @rullzer this fixes the messages being marked as read when loaded in the background, fix https://github.com/owncloud/mail/issues/822 – please also review! :)

File: lib/controller/messagescontroller.php [hunk 1]
```diff
@@ -274,13 +274,20 @@ public function saveAttachment($accountId, $folderId, $messageId, $attachmentId,
...
 	 * @return JSONResponse
 	 */
-	public function toggleStar($accountId, $folderId, $messageId, $starred) {
+	public function setFlags($accountId, $folderId, $messageId, $flags) {
 		$mailBox = $this->getFolder($accountId, $folderId);
 
-		$mailBox->setMessageFlag($messageId, Horde_Imap_Client::FLAG_FLAGGED, !$starred);
+		foreach($flags as $flag => $value) {
+			$value = filter_var($value, FILTER_VALIDATE_BOOLEAN);
+			if ($flag === 'unseen') {
```
Comment (ChristophWurst) [inline lib/controller/messagescontroller.php]: can't we just send 'seen' to the server?

File: js/mail.js [hunk 2]
```diff
@@ -1179,6 +1179,7 @@ var Mail = {
 					$('#forward-button').hide();
 				}
 
+//				Mail.UI.messageView.flagMessage(messageId, 'seen', true);
```
Comment (ChristophWurst) [inline js/mail.js]: I guess this line can be removed ;-)

**Comment by jancborchardt** [comment]
@DeepDiver1975 still got a message marked as read without opening it.

File: lib/controller/messagescontroller.php (hunk 1 above)
Comment (DeepDiver1975) [inline lib/controller/messagescontroller.php]: would require some more changes - but yes in a second pr ..

**Comment by ChristophWurst** [comment]
@DeepDiver1975 @jancborchardt same here, some messages are still flagged as seen before I actually open them

**Comment by jancborchardt** [comment]
@DeepDiver1975 to reproduce it, refresh the page 2–3 times. Then all mails will be marked as read.

**Comment by ChristophWurst** [comment]
Digging into horde.log shows that when the message is fetched, peek mode is used. However, there's a second body fetch command that doesn't peek, hence the flag is set…

**Comment by ChristophWurst** [comment]
Found it: https://github.com/owncloud/mail/blob/master/lib/message.php#L499

We need to peek there too

**Comment by ChristophWurst** [comment]
@jancborchardt I added my fixes. Please test again :)

**Comment by jancborchardt** [comment]
@DeepDiver1975 @irgendwie @PoPoutdoor @Gomez please also check! :)

**Pull request merged**